src/zkevm_specs/evm/execution/oog_call.py

```diff
@@ -0,0 +1,107 @@
+    # retrieve op code associated to oog call error
+    opcode = instruction.opcode_lookup(True)
+    instruction.constrain_equal(opcode, Opcode.CALL)
+
+    tx_id = instruction.call_context_lookup(CallContextFieldTag.TxId)
+    reversion_info = instruction.reversion_info()
+    is_static = instruction.call_context_lookup(CallContextFieldTag.IsStatic)
+    # Lookup values from stack
+    gas_rlc = instruction.stack_pop()
+    callee_address_rlc = instruction.stack_pop()
+    value = instruction.stack_pop()
+    cd_offset_rlc = instruction.stack_pop()
+    cd_length_rlc = instruction.stack_pop()
+    rd_offset_rlc = instruction.stack_pop()
+    rd_length_rlc = instruction.stack_pop()
+    is_success = instruction.stack_push()
+    instruction.constrain_zero(is_success)
+
+    cd_offset, cd_length = instruction.memory_offset_and_length(cd_offset_rlc, cd_length_rlc)
+    rd_offset, rd_length = instruction.memory_offset_and_length(rd_offset_rlc, rd_length_rlc)
+
+    # Verify memory expansion
+    next_memory_size, memory_expansion_gas_cost = instruction.memory_expansion_dynamic_length(
+        cd_offset,
+        cd_length,
+        rd_offset,
+        rd_length,
+    )
+
+    # Check not is_static if call has value
+    has_value = 1 - instruction.is_zero(value)
+    instruction.constrain_zero(has_value * is_static)
+    callee_address = instruction.rlc_to_fq(callee_address_rlc, N_BYTES_ACCOUNT_ADDRESS)
+
+    # TODO: handle PrecompiledContract oog cases
+
+    # Add callee to access list
+    is_warm_access = instruction.add_account_to_access_list(tx_id, callee_address, reversion_info)
+
+    # lookup balance of callee
+    callee_balance = instruction.account_read(callee_address, AccountFieldTag.Balance)
+    # Verify gas cost
+    callee_nonce = instruction.account_read(callee_address, AccountFieldTag.Nonce)
+    callee_code_hash = instruction.account_read(callee_address, AccountFieldTag.CodeHash)
+
+    is_empty_code_hash = instruction.is_equal(
+        callee_code_hash, instruction.rlc_encode(EMPTY_CODE_HASH, 32)
+    )
+    is_account_empty = (
+        instruction.is_zero(callee_nonce) * instruction.is_zero(callee_balance) * is_empty_code_hash
+    )
+    gas_cost = (
+        instruction.select(
+            is_warm_access, FQ(GAS_COST_WARM_ACCESS), FQ(GAS_COST_ACCOUNT_COLD_ACCESS)
+        )
+        + has_value * (GAS_COST_CALL_WITH_VALUE + is_account_empty * GAS_COST_NEW_ACCOUNT)
+        + memory_expansion_gas_cost
+    )
+
+    # verify gas is insufficient
+    gas_not_enough, _ = instruction.compare(instruction.curr.gas_left, gas_cost, N_BYTES_GAS)
+    instruction.constrain_equal(gas_not_enough, FQ(1))
+
+    # current call must be failed.
+    instruction.constrain_equal(
+        instruction.call_context_lookup(CallContextFieldTag.IsSuccess), FQ(0)
+    )
+    is_persistent = instruction.call_context_lookup(CallContextFieldTag.IsPersistent)
+    instruction.constrain_equal(is_persistent, FQ(0))
```

Comment:
By https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/815, we can skip this check right?

Reply:
yes, will remove it .

Reply:
@ChihChengLiang  updated in [8e394ee](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/287/commits/8e394eebb07f32b0fc26e10d4918d835e01575d5) thanks!